AnimatedComponents/src/components/timer/Timer.tsx

```diff
@@ -0,0 +1,191 @@
+    strokeDashoffset: (1 - progress.value) * 2 * Math.PI * 45,
+  }))
+
+  const animatedLinearStyle = useAnimatedStyle(() => ({
+    width: `${progress.value * 100}%`,
+    backgroundColor: linearTimerColor,
+  }))
+
+  useEffect(() => {
+    if (isRunning) {
+      const interval = setInterval(() => {
+        setTimeLeft(prev => (prev > 0 ? prev - 1 : 0))
+      }, 1000)
+      return () => clearInterval(interval)
+    }
+  }, [isRunning])
+
+  useEffect(() => {
+    if (timeLeft === 0) {
+      setIsRunning(false)
+    }
+  }, [timeLeft])
+
+  useEffect(() => {
+    handleControl('start')
+  }, [])
+
+  const handleStart = () => {
+    handleControl('start')
+  }
+
+  const handleResume = () => {
+    handleControl('resume')
+  }
+
+  const handlePause = () => {
+    handleControl('pause')
+  }
+
+  const renderControlButtons = useMemo(
+    () => (
+      <>
+        {!isRunning && !isPaused ? (
+          <TouchableOpacity onPress={handleStart} style={[styles.button, buttonStyles?.container]}>
+            <Text style={[styles.buttonText, buttonStyles?.text]}>Start</Text>
+          </TouchableOpacity>
+        ) : (
+          <TouchableOpacity onPress={handlePause} style={[styles.button, buttonStyles?.container]}>
+            <Text style={[styles.buttonText, buttonStyles?.text]}>Pause</Text>
+          </TouchableOpacity>
+        )}
+        {isPaused && (
+          <TouchableOpacity onPress={handleResume} style={[styles.button, buttonStyles?.container]}>
+            <Text style={[styles.buttonText, buttonStyles?.text]}>Resume</Text>
+          </TouchableOpacity>
+        )}
+        <TouchableOpacity onPress={handleReset} style={[styles.button, buttonStyles?.container]}>
+          <Text style={[styles.buttonText, buttonStyles?.text]}>Reset</Text>
+        </TouchableOpacity>
+      </>
+    ),
+    [isRunning, isPaused, buttonStyles, handleControl, handleReset],
+  )
+
+  const formatTime = (seconds: number) => {
+    const minutes = Math.floor(seconds / 60)
+    const remainingSeconds = seconds % 60
+    return `${String(minutes).padStart(2, '0')}:${String(remainingSeconds).padStart(2, '0')}`
+  }
+
+  const renderTimer = () => {
+    return timerType === TimerPreset.Linear ? (
+      <View style={styles.linearContainer}>
+        <Animated.View style={[styles.linearProgress, animatedLinearStyle]} />
+      </View>
+    ) : (
+      <View style={styles.circularContainer}>
+        <Svg width={100} height={100} viewBox="0 0 100 100">
+          <SvgCircle cx="50" cy="50" r="45" stroke="#e0e0e0" strokeWidth="10" fill="none" />
+          <AnimatedSvgCircle
+            cx="50"
+            cy="50"
+            r="45"
+            stroke={circularTimerStrokeColor}
+            strokeWidth="10"
+            fill="none"
+            strokeDasharray={`${2 * Math.PI * 45}`}
+            animatedProps={animatedCircleProps}
+          />
+        </Svg>
+        <Text style={styles.timeLeftTextCircular}>{formatTime(timeLeft)}</Text>
```

Comment:
make a boolean prop which will control whether to show or not for both linear and circular